polynote-spark/src/main/scala/polynote/kernel/interpreter/python/PySparkInterpreter.scala

```diff
@@ -17,13 +17,18 @@ object PySparkInterpreter {
 
   private lazy val py4jToken: String = RandomStringUtils.randomAlphanumeric(256)
 
-  private lazy val gwBuilder: GatewayServerBuilder = new GatewayServerBuilder()
-    .authToken(py4jToken)
-    .javaPort(0)
-    .callbackClient(0, InetAddress.getByName(GatewayServer.DEFAULT_ADDRESS))
-    .connectTimeout(GatewayServer.DEFAULT_CONNECT_TIMEOUT)
-    .readTimeout(GatewayServer.DEFAULT_READ_TIMEOUT)
-    .customCommands(null)
+  private lazy val gwBuilder: GatewayServerBuilder = {
+    val builder = new GatewayServerBuilder()
+      .javaPort(0)
+      .callbackClient(0, InetAddress.getByName(GatewayServer.DEFAULT_ADDRESS))
+      .connectTimeout(GatewayServer.DEFAULT_CONNECT_TIMEOUT)
+      .readTimeout(GatewayServer.DEFAULT_READ_TIMEOUT)
+      .customCommands(null)
+
+    try builder.authToken(py4jToken) catch {
```

Comment:
This shouldn't fail ever as I'm already providing py 0.10.7.jar in the dependencies inside build.sbt. Let me see if it's possible to force using spark's py4j instead of the one shipped inside polynote jars.

Reply:
Well, py4j is provided by Spark, so it all depends on which version of Spark you have. We can compile and link against 0.10.7, but if you're running Spark 2.1.1 then you have 0.10.4, which doesn't have the `authToken` method and you'll get NoSuchMethodError.